output/DeviceListAuditTool.py

```diff
@@ -0,0 +1,188 @@
+    ACNET_error_file = open('ACNET_no_such_property.txt', 'w')
+    with open(log_filename, 'r') as logfile:
+        for logline_i, log_line in enumerate(logfile):
+            if maxcount > 0 and logline_i > maxcount: break
+            logline = log_line.strip()
+            if logline.count(ACNET_NOT_FOUND) > 0:
+                # Extract the last word, hopefully a device name
+                logdevice = logline.split(' ')[-1]
+                if logdevice not in device_error_tallies.keys():
+                    ACNET_error_file.write(logdevice+'\n')
+                    device_error_tallies[logdevice] = 1
+                    if logdevice.startswith('Z'): device_error_tally_Z += 1
+                else: device_error_tallies[logdevice] += 1
+    # Summarize some analysis of the devices garnering these error codes
+    ACNET_error_file.close()
+    print (f'\n__Analysis of nanny log file__ :')
+    # Sort by tallies, highest to lowest
+    device_error_tallies = dict(sorted(device_error_tallies.items(), key=lambda item: item[1], reverse=True))
+
+    # OK, I should have done this in a Pandas dataframe.  I'm sorry.
+    # Make a list of devices having each error count
+    errorcounts_dict = {}
+    for logdevice, errorcount in device_error_tallies.items():
+        if not errorcount in errorcounts_dict.keys(): errorcounts_dict[errorcount] = []
+        errorcounts_dict[errorcount].append(logdevice)
+    # Output a digest of the error counts and how many devices have each error count
+    for errorcount, devicelist in  errorcounts_dict.items():
+        ErrorDevices_haystack = sum(1 for errordevicename in devicelist if errordevicename in haystack_devices)
+        print (f'{len(devicelist)} devices in the logger file EACH had {errorcount} instances of "{ACNET_NOT_FOUND}".')
+        print (f'({device_error_tally_Z} devices with {ACNET_NOT_FOUND} begin with Z)')
+        print (f'   and {ErrorDevices_haystack} of these device names were found in {DRF_file}')
+        if needle_filename != '':
+            ErrorDevices_needlestack = sum(1 for errordevicename in devicelist if errordevicename in needles_found)
+            print (f'    and {ErrorDevices_needlestack} devices with this error code were found in {needle_filename}')
+
+proposed_to_drop = []
+# Generate a suggested, edited DRF file, and a summary table
+AllGood                                   = []
+ZDevices___________                       = []
+NonSettingsToIgnore                       = []
+NoSuchProperty16_13                       = []
+ZDevices___________ANDNonSettingsToIgnore = []
+ZDevices___________ANDNoSuchProperty16_13 = []
+NonSettingsToIgnoreANDNoSuchProperty16_13 = []
+ZDevices___________ANDNonSettingsToIgnoreANDNoSuchProperty16_13 = []
+devicestodrop = []
+uncounted = 0
+with open(DRF_file, 'r') as DRF_lines, open(DRF_file+'NEW', 'w') as newfile:
+    for linecount, drf_line in enumerate(DRF_lines):
+        if maxcount > 0 and linecount > maxcount: break
```

Comment:
This has the same problems as it's sibling above.

Reply:
It's fixed just like its subling. It's a sibling, is what it is.